pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_partial_index failed [TransactionRetryWithProtoRefreshError on index creation]

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_partial_index [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/15943217?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/15943217?buildTab=artifacts#/) on master @ [295d09a88895a69e5cc9149fb8165acf78e39e61](https://github.com/cockroachdb/cockroach/commits/295d09a88895a69e5cc9149fb8165acf78e39e61):


```
=== RUN   TestTenantLogic_partial_index
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f82bd4d116da0800a539c75f49940d6f/logTestTenantLogic_partial_index3175470877
    test_log_scope.go:81: use -show-logs to present logs inline
[05:36:23] setting distsql_workmem='17395B';
[05:36:23] setting distsql_workmem='17395B';
[05:36:25] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4074/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index: 82 statements
[05:36:31] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4074/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index with config 3node-tenant: 99 tests, 0 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4074/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: error while processing
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4074/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=a8969e98 key=/Tenant/10/Table/7/1/0/0 iso=Serializable pri=0.06129593 epo=0 ts=1720157788.872089549,2 min=1720157786.829626719,0 seq=28} lock=true stat=ABORTED rts=1720157786.829626719,0 wto=false gul=1720157787.329626719,0
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82bd4d116da0800a539c75f49940d6f/logTestTenantLogic_partial_index3175470877
--- FAIL: TestTenantLogic_partial_index (11.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_partial_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40112

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_partial_index [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/15962701?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/15962701?buildTab=artifacts#/) on master @ [485975b3a824c68c07340e6a336c7864c00d3c6d](https://github.com/cockroachdb/cockroach/commits/485975b3a824c68c07340e6a336c7864c00d3c6d):


```
=== RUN   TestTenantLogic_partial_index
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f82bd4d116da0800a539c75f49940d6f/logTestTenantLogic_partial_index943042731
    test_log_scope.go:81: use -show-logs to present logs inline
[05:37:33] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4076/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index: 58 statements
[05:37:39] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4076/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index with config 3node-tenant: 99 tests, 0 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4076/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: error while processing
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4076/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "sql txn" meta={id=bca53adc key=/Tenant/10/Table/7/1/0/0 iso=Serializable pri=0.02161860 epo=0 ts=1720330654.386492632,0 min=1720330654.386492632,0 seq=36} lock=true stat=ABORTED rts=1720330654.386492632,0 wto=false gul=1720330654.886492632,0
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82bd4d116da0800a539c75f49940d6f/logTestTenantLogic_partial_index943042731
--- FAIL: TestTenantLogic_partial_index (11.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_partial_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_partial_index [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/15971634?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/15971634?buildTab=artifacts#/) on master @ [485975b3a824c68c07340e6a336c7864c00d3c6d](https://github.com/cockroachdb/cockroach/commits/485975b3a824c68c07340e6a336c7864c00d3c6d):


```
=== RUN   TestTenantLogic_partial_index
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/f82bd4d116da0800a539c75f49940d6f/logTestTenantLogic_partial_index2024852354
    test_log_scope.go:81: use -show-logs to present logs inline
[05:35:34] setting distsql_workmem='55512B';
[05:35:34] setting distsql_workmem='55512B';
[05:35:36] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4069/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index: 78 statements
[05:35:43] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4069/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index with config 3node-tenant: 99 tests, 0 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4069/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: error while processing
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4069/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=161c3cea key=/Tenant/10/Table/7/1/0/0 iso=Serializable pri=0.00467417 epo=0 ts=1720416937.910588096,0 min=1720416937.910588096,0 seq=33} lock=true stat=PENDING rts=1720416937.910588096,0 wto=false gul=1720416938.410588096,0
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f82bd4d116da0800a539c75f49940d6f/logTestTenantLogic_partial_index2024852354
--- FAIL: TestTenantLogic_partial_index (11.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_partial_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_partial_index [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17415405?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17415405?buildTab=artifacts#/) on master @ [b13f76063a555af796f5df7243ecdd894ff0ff51](https://github.com/cockroachdb/cockroach/commits/b13f76063a555af796f5df7243ecdd894ff0ff51):


```
=== RUN   TestTenantLogic_partial_index
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/6b89db816bf074ebea6fff85a8db7616/logTestTenantLogic_partial_index1883979
    test_log_scope.go:76: use -show-logs to present logs inline
[20:44:27] setting distsql_workmem='79082B';
[20:44:27] setting distsql_workmem='79082B';
[20:44:30] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16193/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index: 83 statements
[20:44:36] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16193/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index with config 3node-tenant: 99 tests, 0 failures
    logic.go:4229: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16193/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: error while processing
    logic.go:4229: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16193/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/partial_index:581: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=97f5d411 key=/Tenant/10/Table/7/1/0/0 iso=Serializable pri=0.01370334 epo=0 ts=1729629870.928312511,1 min=1729629870.853385993,0 seq=27} lock=true stat=PENDING rts=1729629870.853385993,0 wto=false gul=1729629871.353385993,0
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6b89db816bf074ebea6fff85a8db7616/logTestTenantLogic_partial_index1883979
--- FAIL: TestTenantLogic_partial_index (10.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_partial_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/partial_index
+++ b/pkg/sql/logictest/testdata/logic_test/partial_index
@@ -564,10 +564,11 @@ SELECT * FROM h@a_b_foo_idx WHERE b = 'foo'
 1  foo
 
 # Backfill a partial index with a user defined type when a new table is created
-# in the same transaction.
+# in the same transaction. Use a high priority to make it less likely for the
+# transaction to be aborted.
 
 statement ok
-BEGIN
+BEGIN PRIORITY HIGH
 
 statement ok
 CREATE TABLE i (a INT, b enum)

--- a/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
+++ b/pkg/sql/logictest/testdata/logic_test/synthetic_privileges
@@ -296,8 +296,9 @@ SELECT has_table_privilege('testuser4', 'crdb_internal.feature_usage', 'SELECT')
 ----
 false
 
+# Use a high priority transaction to avoid this transaction being aborted.
 statement ok
-BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
+BEGIN TRANSACTION PRIORITY HIGH ISOLATION LEVEL SERIALIZABLE;
 
 statement ok
 GRANT SELECT ON crdb_internal.feature_usage TO testuser4

